app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExpectToExceedThresholdWeightController.scala

```diff
@@ -35,43 +36,46 @@ class ExpectToExceedThresholdWeightController @Inject() (
                                                           journeyAction: JourneyAction,
                                                           override val registrationConnector: RegistrationConnector,
                                                           mcc: MessagesControllerComponents,
-                                                          page: expect_to_exceed_threshold_weight_page
+                                                          page: expect_to_exceed_threshold_weight_page,
+                                                          form: ExpectToExceedThresholdWeight
+
+
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with Cacheable with I18nSupport {
 
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction) { implicit request =>
-      request.registration.liabilityDetails.expectToExceedThresholdWeight match {
-        case Some(data) =>
-          Ok(page(ExpectToExceedThresholdWeight.form().fill(data)))
-        case _ => Ok(page(ExpectToExceedThresholdWeight.form()))
+      (request.registration.liabilityDetails.expectToExceedThresholdWeight, request.registration.liabilityDetails.dateRealisedExpectedToExceedThresholdWeight) match {
+        case (Some(yesNo), date) => Ok(page(form().fill(ExpectToExceedThresholdWeightAnswer(yesNo,date.map(_.date)))))
+        case _ => Ok(page(form()))
       }
     }
 
   def submit(): Action[AnyContent] =
     (authenticate andThen journeyAction).async { implicit request =>
-      ExpectToExceedThresholdWeight.form()
+      form()
         .bindFromRequest()
         .fold(formWithErrors => Future.successful(BadRequest(page(formWithErrors))),
               expectToExceed =>
                 updateRegistration(expectToExceed).map {
-                  case Right(_) =>
-                    if (expectToExceed)
-                      Redirect(routes.ExpectToExceedThresholdWeightDateController.displayPage())
-                    else Redirect(routes.NotLiableController.displayPage())
+                  case Right(_) =>  Redirect(routes.TaxStartDateController.displayPage())
+//                    if (expectToExceed)
```

Comment:
Same here. Are we at a stage where we can update this routing? Or remove the commented out code?